### PR TITLE
bgpv2: Render route policies for LB VIPs instead of pools

### PR DESCRIPTION
**Pull request by rastislavs**
BGPv2 API required `Service` advertisement selector to match both `Service` as well as `CiliumLoadBalancerIPPool` from which the service got assigned the LB VIP. 

This was not documented, [confused some users](https://github.com/cilium/cilium/issues/33076#issuecomment-2262844827), and could not work properly in some cases (large pool for many services, from which only some should be advertised). To fix all of that, we are removing the requirement to match the pool and render route-policies per LB VIP instead of per LB pool.

```release-note
BGPv2: Fix `Service` advertisement selector: do not require matching `CiliumLoadBalancerIPPool`
```

### CODE REVIEW

File: pkg/bgpv1/manager/reconcilerv2/service.go [hunk 1]
```diff
@@ -358,24 +271,49 @@ func (r *ServiceReconciler) getDiffRoutePolicies(p ReconcileParams, toUpdate []*
...
+			for _, advert := range adverts {
+				labelSelector, err := slim_metav1.LabelSelectorAsSelector(advert.Selector)
+				if err != nil {
+					return nil, fmt.Errorf("failed constructing LabelSelector: %w", err)
+				}
+				if !labelSelector.Matches(serviceLabelSet(svc)) {
+					continue
+				}
+				// LoadBalancerIP
+				policy, err := r.getLoadBalancerIPRoutePolicy(p, peer, agentFamily, svc, advert, ls)
```
Comment (harsimran-pabla) [inline pkg/bgpv1/manager/reconcilerv2/service.go]: The single variable definition can cause some confusion and pointer reuse issues. Maybe use explicit policy definition names: lbPolicy, extPolicy, clusterPolicy.

**Review by harsimran-pabla — COMMENTED** [review]
Minor change in variable naming. Otherwise, very cool change. 

I think it should be backported to 1.16 as well.

File: pkg/bgpv1/manager/reconcilerv2/service.go (hunk 1 above)
Comment (rastislavs) [reply pkg/bgpv1/manager/reconcilerv2/service.go]: ack, done

**Review by harsimran-pabla — APPROVED** [review]

**Pull request merged**